nodes/object/object_visibility_output.py

```diff
@@ -0,0 +1,160 @@
+
+    def create(self):
+        self.inputs.new("an_ObjectSocket", "Object", "object").defaultDrawType = "PROPERTY_ONLY"
+        self.inputs.new("an_BooleanSocket", "Hide", "hide")
+        self.inputs.new("an_BooleanSocket", "Hide Select", "hideSelect")
+        self.inputs.new("an_BooleanSocket", "Hide Render", "hideRender")
+        self.inputs.new("an_BooleanSocket", "Show Name", "showName")
+        self.inputs.new("an_BooleanSocket", "Show Axis", "showAxis")
+        self.inputs.new("an_BooleanSocket", "Show Xray", "showXray")
+        self.outputs.new("an_ObjectSocket", "Object", "object")
+        for socket in self.inputs: socket.value = False
+        self.updateSocketVisibility()
+        
+    def draw(self, layout):
+        row = layout.row()
+        
+        col = row.column()
+        subrow = col.row(align = True)
+        subrow.alignment = 'LEFT'
+        #row.label("Visibility")
+        subrow.prop(self, "useView", text = "", icon = "RESTRICT_VIEW_OFF")
+        subrow.prop(self, "useSelect", text = "", icon = "RESTRICT_SELECT_OFF")
+        subrow.prop(self, "useRender", text = "", icon = "RESTRICT_RENDER_OFF")
+        
+        col = row.column()
+        subrow = col.row(align = True)
+        subrow.alignment = 'RIGHT'
+        #row.label("Show")
+        subrow.prop(self, "useName", text = "", icon = "SORTALPHA")    # SYNTAX_ON / SYNTAX_OFF
+        subrow.prop(self, "useAxis", text = "", icon = "MANIPUL")    
+        subrow.prop(self, "useXray", text = "", icon = "ROTACTIVE")    # MOD_DECIM / META_BALL
+    
+    def drawAdvanced(self, layout):
+        layout.operator("wm.call_menu", text = "Node Info / Help", icon = "INFO").name = "an.show_object_visibility_output_help"
+
+    def updateSocketVisibility(self):
+        self.inputs["Hide"].hide = not (self.useView)
+        self.inputs["Hide Select"].hide = not (self.useSelect)
+        self.inputs["Hide Render"].hide = not (self.useRender)
+        self.inputs["Show Name"].hide = not (self.useName)
+        self.inputs["Show Axis"].hide = not (self.useAxis)
+        self.inputs["Show Xray"].hide = not (self.useXray)
+        
+    def getExecutionCode(self):
+        lines = []
+        lines.append("if object is not None:")
+        
+        if self.useView: lines.append("    object.hide = hide")
+        if self.useSelect: lines.append("    object.hide_select = hideSelect")
+        if self.useRender: lines.append("    object.hide_render = hideRender")
+        
+        if self.useName: lines.append("    object.show_name = showName")
+        if self.useAxis: lines.append("    object.show_axis = showAxis")
+        if self.useXray: lines.append("    object.show_x_ray = showXray")
+        
+        if not any((self.useView, self.useSelect, self.useRender, self.useName, self.useAxis, self.useXray)): 
```

Comment:
this check shall come first in this function. So that you immediatly return an empty list when nothing is used